server/src/main/java/org/opensearch/cluster/routing/remote/InternalRemoteRoutingTableService.java

```diff
@@ -268,6 +272,53 @@ private void uploadIndex(
         }
     }
 
+
+    @Override
+    public CheckedRunnable<IOException> getAsyncIndexRoutingReadAction(
+        String uploadedFilename,
+        Index index,
+        LatchedActionListener<IndexRoutingTable> latchedActionListener) {
+        int idx = uploadedFilename.lastIndexOf("/");
+        String blobFileName = uploadedFilename.substring(idx+1);
+        BlobContainer blobContainer = blobStoreRepository.blobStore().blobContainer(BlobPath.cleanPath().add(uploadedFilename.substring(0,idx)));
+
+        return () -> readAsync(
+            blobContainer,
+            blobFileName,
+            index,
+            threadPool.executor(ThreadPool.Names.REMOTE_STATE_READ),
+            ActionListener.wrap(response -> latchedActionListener.onResponse(response.getIndexRoutingTable()), latchedActionListener::onFailure)
+        );
+    }
+
+    private void readAsync(BlobContainer blobContainer, String name, Index index, ExecutorService executorService, ActionListener<RemoteIndexRoutingTable> listener) {
+        executorService.execute(() -> {
+            try {
+                listener.onResponse(read(blobContainer, name, index));
+            } catch (Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private RemoteIndexRoutingTable read(BlobContainer blobContainer, String path, Index index) {
+        try {
+            return new RemoteIndexRoutingTable(blobContainer.readBlob(path), index);
+        } catch (IOException | AssertionError e) {
+            logger.info("RoutingTable read failed with error: {}", e.toString());
```

Comment:
This should be `logger.error`

Reply:
Also we dont need `e.toString()`, we can add exception directly here. Logger handles the conversions required by default.

Reply:
Addressed.